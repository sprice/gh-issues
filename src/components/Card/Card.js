import React from "react"
import PropTypes from "prop-types"

class Card extends React.Component {
  static Icons = ({ isClosed, isPullRequest }) => (
    <div>
      {isClosed ? <i className="icon closed" /> : null}
      {isPullRequest ? <i className="icon pull-request" /> : null}
    </div>
  )

  static Title = ({ children }) => (
    <div className="card-title">
      <h4 className="regular">{children}</h4>
    </div>
  )
  static Body = ({ children }) => (
    <div className="card-body">
      <p>{children || "No description provided."}</p>
    </div>
  )
  static Label = ({ children }) => (
    <small className="card-label light">&#8226; {children}</small>
  )

  render() {
    return (
      <div className="card">
        {React.Children.map(this.props.children, child =>
          React.cloneElement(child, {
            isClosed: this.props.isClosed,
            isPullRequest: this.props.isPullRequest
          })
        )}
      </div>
    )
  }
}

Card.propTypes = {
  children: PropTypes.any.isRequired,
  isClosed: PropTypes.bool,
  isPullRequest: PropTypes.bool
}

export default Card
