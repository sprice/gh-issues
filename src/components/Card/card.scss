.cards {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  max-width: $breakpoint-x-large;
  margin: 0 auto;
}

.card {
  position: relative;
  width: 100%;
  min-width: 200px;
  max-width: 20rem;
  margin-bottom: 3rem;
  margin-right: 1.5rem;
  margin-left: 1.5rem;
  padding-bottom: 3rem;
  box-shadow: 0px 0px 10px 2px rgba($grey, 0.4);
}

i.closed::before {
  content: "";
  display: block;
  width: 10px;
  height: 10px;
  background: url("../icons/issue-closed.svg") no-repeat;
}

i.pull-request::before {
  content: "";
  display: block;
  width: 10px;
  height: 10px;
  background: url("../icons/pull-request.svg") no-repeat;
}

.card-title {
  background-color: $light-blue;
  color: $white;
  padding: 1rem;
  word-break: break-all;
}

.card-title h4 {
  clear: both;
}

.card-body {
  color: $grey;
  padding: 0.5rem;
  word-break: break-all;
}

.card-label {
  position: absolute;
  bottom: 0.5rem;
  display: inline-block;
  background-color: $pink;
  color: $white;
  margin-left: 0.5rem;
  padding: 0.375rem;
  border-radius: 5px;
}

@media (min-width: $breakpoint-small) {
  .card-title {
    word-break: inherit;
  }
  .card-body {
    word-break: inherit;
  }
}
