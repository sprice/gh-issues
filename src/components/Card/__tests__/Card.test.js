import React from "react"
import { create } from "react-test-renderer"
import Card from "../Card"

test("Snapshop of Card", () => {
  const labels = [
    {
      id: 42,
      name: "awesome"
    },
    {
      id: 21000000,
      name: "sauce"
    }
  ]
  const c = create(
    <Card isClosed={true} isPullRequest={true}>
      <Card.Icons />
      <Card.Title>Issue Title</Card.Title>
      <Card.Body>Issue Body</Card.Body>
      {labels.map(label => (
        <Card.Label key={label.id}>{label.name}</Card.Label>
      ))}
    </Card>
  )
  expect(c.toJSON()).toMatchSnapshot()
})
