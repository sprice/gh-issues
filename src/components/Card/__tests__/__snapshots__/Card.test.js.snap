// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshop of Card 1`] = `
<div
  className="card"
>
  <div>
    <i
      className="icon closed"
    />
    <i
      className="icon pull-request"
    />
  </div>
  <div
    className="card-title"
  >
    <h4
      className="regular"
    >
      Issue Title
    </h4>
  </div>
  <div
    className="card-body"
  >
    <p>
      Issue Body
    </p>
  </div>
  <small
    className="card-label light"
  >
    • 
    awesome
  </small>
  <small
    className="card-label light"
  >
    • 
    sauce
  </small>
</div>
`;
